src/cursor/cur_table.c

```diff
@@ -988,6 +988,13 @@ __wt_curtable_open(WT_SESSION_IMPL *session,
 		    table->cgroups[0]->source, NULL, cfg, cursorp);
 
 		__wt_schema_release_table(session, table);
+		if (ret == 0) {
+			/* Fix up the public URI to match what was passed in. */
+			cursor = *cursorp;
+			__wt_free(session, cursor->uri);
+			cursor->uri = NULL;
```

Comment:
@michaelcahill, thanks.  I've fixed that, and also another issue with LSM ownership of the `WT_CURSOR->uri` field.

Can you take another look please?